src/common/config_validators.cc

```diff
@@ -0,0 +1,57 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "common/config_validators.h"
+#include "include/stringify.h"
+#include "include/rbd/features.h"
+#include <map>
+#include <vector>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+int validate(md_config_t::option_rbd_default_features_t *,
+             std::string *value, std::string *error_message) {
+  static const std::map<std::string, uint64_t> FEATURE_MAP = {
+    {RBD_FEATURE_NAME_LAYERING, RBD_FEATURE_LAYERING},
+    {RBD_FEATURE_NAME_STRIPINGV2, RBD_FEATURE_STRIPINGV2},
+    {RBD_FEATURE_NAME_EXCLUSIVE_LOCK, RBD_FEATURE_EXCLUSIVE_LOCK},
+    {RBD_FEATURE_NAME_OBJECT_MAP, RBD_FEATURE_OBJECT_MAP},
+    {RBD_FEATURE_NAME_FAST_DIFF, RBD_FEATURE_FAST_DIFF},
+    {RBD_FEATURE_NAME_DEEP_FLATTEN, RBD_FEATURE_DEEP_FLATTEN},
+    {RBD_FEATURE_NAME_JOURNALING, RBD_FEATURE_JOURNALING},
+    {RBD_FEATURE_NAME_DATA_POOL, RBD_FEATURE_DATA_POOL},
+  };
```

Comment:
May be add a static assert to make sure the map is updated when a new feature is added, something like below?
```
static_assert((RBD_FEATURE_DATA_POOL << 1) > RBD_FEATURES_ALL, "new feature added");
```